### ISSUE TITLE
accounts/keystore: doesn't see my key file

### ISSUE THREAD

**Issue by ganchix** [ethereum/go-ethereum#15176]
#### System information

Geth version: `v1.7.0-unstable`
OS & Version: Linux (Latest ethereum/client-go docker image)

#### Expected behaviour

When a wallet file is moved to keystore must refresh the account list without rebooting the node.
#### Actual behaviour

The file is moved but the account list doesn't change.
Maybe there is a problem with [wath.go
](https://github.com/ethereum/go-ethereum/blob/master/accounts/keystore/watch.go)

#### Steps to reproduce the behaviour

1. Start your geth node.
2. Move your wallet file to keystore
3. Attach the node.
4. eth.accounts doesn't have the new account

**Comment by mexskican** [comment]
It is working for me,
I am using the current go-ethereum branch master on ubuntu 16.04.
Can you show how you start your node and how you attach to the node.

![account-bug](https://user-images.githubusercontent.com/12745076/30722370-bc9d55d8-9ee5-11e7-99cf-a0bb08845f13.png)

**Steps to reproduce the behaviour**
```
$ nohup geth --verbosity 0 >> geth.log &
// Add a new keyfile to the keystore
$ cp PathToYourKeyFile ~/.ethereum/keystore
$ geth attach
>eth.accounts
```

**Comment by ganchix** [comment]
Ok, I think that my problem is other, I am using a [s3fs](https://github.com/s3fs-fuse/s3fs-fuse) (A filesystem syncronized by S3 Amazon service) to syncronize my keystore folder.

My problem happen when I upload the wallet file to s3 and s3fs syncronize the folder but the geth node doesn't see the new account.

**Comment by ganchix** [comment]
How to do it, 

I syncronize my wallet folder 


Later I start my geth node 

`geth --datadir /home/rafa/ethereum/`

After

**Comment by ganchix** [comment]
This is a part of my script : 

```
export AWSACCESSKEYID=${AWSACCESSKEYID:-$AWS_ACCESS_KEY_ID}
export AWSSECRETACCESSKEY=${AWSSECRETACCESSKEY:-$AWS_SECRET_ACCESS_KEY}
s3fs ${S3_BUCKET_KEYSTORE} /geth/.geth/keystore
....
geth --datadir /geth/.geth \
        --nodiscover \
        --unlock $MINER_ADDRESS --password /geth/password.txt \
        --identity $NODE_NAME \
        --rpccorsdomain="*"  \
        --networkid $NETWORKID \
        --rpcaddr="0.0.0.0" \
        --rpcapi "db,personal,admin,eth,net,web3" \
        --rpc \
        $GETHARGS 2>&1
```

**Comment by mexskican** [comment]
I have never used s3fs but your script looks good to me. 
Can you try to remove and add manually your key file while your running your node, to determine if the problem comes from the way you are launching your node or from your upload with s3fs.

**Comment by Habol** [comment]
Hi Benoit

thanks for your advise/suggestion. But…. what do you mean with s3fs? 
Another question >>> the key file is the one beginning with Utc—2017 etc. etc in the keystore folder?

Best regards from the Netherlands

Han

**Comment by ganchix** [comment]
My geth command is `geth--datadir/geth/.geth --nodiscover --unlock 0 --password /geth/password.txt --identity tx-node-3830646593-d9vgz --rpccorsdomain=* --networkid 888 --rpcaddr=0.0.0.0 --rpcapi "db,personal,admin,eth,net,web3" --rpc`

@mexskican when I copy the keystore it works! But when I upload it to s3 and synchronized the folder it doesn't work.

@Habol s3fs is a filesystem syncronized by S3 Amazon service, all files are generated by geth with `personal.newAccount `method, have the common signature.

**Comment by ganchix** [comment]
I am going to explain my complete use case, I have two geth nodes in two different docker containers deployed it in kubernetes, every web3 request is done in a kubernetes service this means that the request are balanced beetween nodes.

This have several implication, the containers are inmutables, I can't access to do a mv to move files, and the most important is the following: 

 - I did a first request to create a new account, it's is done in the first node, the wallet is created in the first node.

 - Later I did new one request to get the balance of this account, but it's done in the second node, I don't have the wallet file in this node, it never gonna work propertly.

For this reason, I use s3fs, becuase the first node write in the keystore folder and this folder is syncronized with the all nodes, and this problem dissapear. We are considering use Ipfs with fuse but first we want test it with s3fs.
